src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1015,8 +1015,14 @@ def vmss_set(**kwargs):
 
 def update_acs(resource_group_name, container_service_name, new_agent_count):
     client = _compute_client_factory()
-    instance = client.container_service.get(resource_group_name, container_service_name)
+    instance = client.container_services.get(resource_group_name, container_service_name)
     instance.agent_pool_profiles[0].count = new_agent_count
-    return client.container_service.create_or_update(resource_group_name,
-                                                     container_service_name, instance)
-
+    return client.container_services.create_or_update(resource_group_name,
+                                                      container_service_name, instance)
+
+def list_container_services(client, resource_group_name=None):
+    ''' List Container Services. '''
+    svc_list = client.container_services.list_by_resource_group(
```

Comment:
With your change at `commands.py` to reuse the `factory` object, the `client` now points to `container_services` already, so this line might be broken. Please double check. Also it won't hurt to invoke the `list` command in the e2e test case.
I usually avoid passing in operation_group object as `client` to custom commands, because of the inconsistency, hence confusing.